client/signup/steps/launch-site/index.jsx

```diff
@@ -0,0 +1,52 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import SignupActions from 'lib/signup/actions';
+import { launchSite } from 'state/sites/launch/actions';
+
+class LaunchSiteComponent extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			siteId: props.siteId,
+		};
+	}
+
+	componentDidMount() {
+		this.props.launchPrivateSite( this.props.siteId );
+	}
+
+	render() {
+		return null;
+	}
+}
+
+export default connect(
+	null,
+	( dispatch, ownProps ) => ( {
+		launchPrivateSite: () => {
+			const { flowName, stepName, stepSectionName, goToNextStep } = ownProps;
+
+			dispatch( launchSite( ownProps.signupDependencies.siteId ) );
```

Comment:
Would it make more sense to make this the `apiRequestFunction` of the step, since that's what's happening behind the scenes?

Reply:
Agree. Implemented in [d21ca5a](https://github.com/Automattic/wp-calypso/pull/30242/commits/d21ca5a8f0eb32b50a2c80e50a730d494a1bfc21).